zhaquirks/smarthjemmet/quadzigsw.py

```diff
@@ -0,0 +1,87 @@
+"""Device handler for smarthjemmet.dk QUAD-ZIG-SW."""
+
+from zigpy.quirks import CustomCluster
+from zigpy.quirks.v2 import QuirkBuilder
+from zigpy.zcl import ClusterType
+from zigpy.zcl.clusters.general import MultistateInput
+
+from zhaquirks import PowerConfigurationCluster
+from zhaquirks.const import (
+    COMMAND,
+    COMMAND_DOUBLE,
+    COMMAND_HOLD,
+    COMMAND_RELEASE,
+    COMMAND_SINGLE,
+    COMMAND_TRIPLE,
+    DOUBLE_PRESS,
+    ENDPOINT_ID,
+    LONG_PRESS,
+    LONG_RELEASE,
+    SHORT_PRESS,
+    TRIPLE_PRESS,
+    VALUE,
+    ZHA_SEND_EVENT,
+)
+
+MANUFACTURER = "smarthjemmet.dk"
+MODEL_ID = "QUAD-ZIG-SW"
+
+ACTION_TYPE = {
+    0: COMMAND_RELEASE,
+    1: COMMAND_SINGLE,
+    2: COMMAND_DOUBLE,
+    3: COMMAND_TRIPLE,
+    4: COMMAND_HOLD,
+}
+
+
+class CR2032PowerConfigurationCluster(PowerConfigurationCluster):
+    """CR2032 Power Configuration Cluster."""
+
+    MIN_VOLTS = 2.2
+    MAX_VOLTS = 3.0
+
+
+class CustomMultistateInputCluster(CustomCluster, MultistateInput):
+    """Multistate input cluster."""
+
+    def _update_attribute(self, attrid, value):
+        super()._update_attribute(attrid, value)
+        if attrid == MultistateInput.AttributeDefs.present_value.id:
+            if action := ACTION_TYPE.get(value) is not None:
+                event_args = {VALUE: value}
+                self.listener_event(ZHA_SEND_EVENT, action, event_args)
+
+            super()._update_attribute(0, action)
```

Comment:
If the changes work, please also try and see what these additional changes to this line do:
- replacing 0 with `MultistateInput.AttributeDefs.present_value.id` in this line (sensor entity in HA?)
- completely removing this line

I believe this was copied from other quirks where it doesn't actually work, as there is no attribute with `id=0` on this cluster.